vbfml/plot/util.py

```diff
@@ -172,19 +173,75 @@ def plot_histograms_for_each_label(
         data[data["labels"] == 0][variable],
         bins=50,
         density=True,
+        weights=data[data["labels"] == 0]["weights"],
         histtype="step",
         label="background",
     )
     plt.hist(
         data[data["labels"] == 1][variable],
         bins=50,
         density=True,
+        weights=data[data["labels"] == 1]["weights"],
         histtype="step",
         label="signal",
     )
-    plt.axvline(x=cut, color="k", linestyle="--", label=f"cut at {cut:.3f}")
-    plt.title(f"{variable} distribution")
+    if cut:
+        plt.axvline(x=cut, color="k", linestyle="--", label=f"cut at {cut:.3f}")
+    plt.title(f"{variable} distribution{save_name}")
     plt.xlabel(variable)
     plt.ylabel("density counts")
-    plt.legend()
-    plt.savefig(pjoin(outdir, f"{variable}_density.pdf"))
+    plt.ylim(bottom=0)
+    if variable == "score":
+        plt.legend(loc="upper center")
+    else:
+        plt.legend()
+    plt.savefig(pjoin(outdir, f"{variable}_density{save_name}.pdf"))
+
+
+def sort_by_pair_along_first_axis(
+    x: list,
+    y: list,
+    z: list = [None],
+    reverse: bool = False,
+    abs_val: bool = False,
+):
+    """
+    take 2(3) lists/np.array of the same length representing x-y(-z) coordinate-like objects, y_i(-z_i) need to stay with x_i
+    sort them in the good order according to x from lowest to highest (reverse=True for high to low)
+    return 2(3) np.arrays with x sorted, y(-z) got the same permutation as x
+    /!/ sort by absolute values !
```

Comment:
Thanks for the PR @pmayenco. This function mostly looks good, I just have a few suggestions:
- You can set the default value of `z` to be `None` directly. This way, you don't need a separate `z_flag`, if you do `if z:` that will directly evaluate to `True` or `False`
- Can we make the name of `abs_val` parameter something like `sort_by_abs`, so that it's more clear? 
- For this function to work, the three arrays must be of the same length, right? Maybe in the beginning of the function, could you add an assertion checking that this is the case?

Reply:
Okay, if it doesn't work easily I think we can leave it as it is then, thanks for checking. But could you try to implement the other two changes, or let me know if you have any other suggestions about them, thanks!

Reply:
done